src/librustc_session/lint/builtin.rs

```diff
@@ -46,6 +46,18 @@ declare_lint! {
     "shift exceeds the type's number of bits"
 }
 
+declare_lint! {
+    pub OVERFLOW,
+    Deny,
+    "arithmetic operation overflows"
+}
+
+declare_lint! {
+    pub PANIC,
```

Comment:
That sounds like `panic!("...")`, which is indeed unconditional but doesn't fit.

Under the hood this is an assertion failure, but that might also be confusing because it sounds like `assert!`.

Reply:
So it's an "always failing dynamic check". No idea how to name this nicely

Reply:
Despite my reservations, `unconditional_panic` is my favorite so far.